utils/filepattern-generator-plugin/plugin.json

```diff
@@ -0,0 +1,86 @@
+{
+  "name": "Filepattern Generator",
+  "version": "0.1.2",
+  "title": "Filepattern Generator",
+  "description": "Filepattern Generator plugin creates a csv or feather file containing a number of new filepatterns, where each filepattern will subset the data in the directory",
+  "author": "Nick Schaub (nick.schaub@nih.gov), Hamdah Shafqat Abbasi (hamdah.abbasi@axleinfo.com)",
+  "institution": "National Center for Advancing Translational Sciences, National Institutes of Health",
+  "repository": "https://github.com/PolusAI/polus-plugins",
+  "website": "https://ncats.nih.gov/preclinical/core/informatics",
+  "citation": "",
+  "containerId": "polusai/filepattern-generator-plugin:0.1.2",
+  "inputs": [
+    {
+      "name": "inpDir",
+      "type": "collection",
+      "description": "Input image collection to be processed by this plugin",
+      "required": true
+    },
+    {
+      "name": "pattern",
+      "type": "string",
+      "description": "Filepattern to parse image files",
+      "required": false
+    },
+    {
+      "name": "chunkSize",
+      "type": "number",
+      "description": "Number of images to generate collective filepattern",
+      "required": false
+    },
+    {
+      "name": "groupBy",
+      "type": "string",
+      "description": "Select a parameter to generate filepatterns in specific order",
+      "required": false
+    },
+    {
+      "name": "outFormat",
+      "type": "enum",
+      "description": "Output Format of collective filepatterns. Only Supports (CSV and feather) file format. (default file format is CSV)",
+      "required": false,
+      "options" : {
+        "values" : [
+          "csv",
+          "feather"  
+        ]
+      }
+      
+    }
+
+  ],
+  "outputs": [
```

Comment:
If we output a feather file, is it appropriate to store this as a csv collection?

Reply:
Kevin, I changed output collection to genericData. Now both csv and feather file would be saved in genericData on WIPP. It makes more sense to use it since we have two different formats and possibly in future if we want to add other formats then this is more sensible approach in a longer run